File: examples/java/kora-java-validation/src/main/java/io/koraframework/example/validation/ResultValidator.java
package io.koraframework.example.validation;

import org.jspecify.annotations.Nullable;
import java.util.UUID;
import io.koraframework.common.annotation.Component;
import io.koraframework.common.annotation.Root;
import io.koraframework.validation.common.annotation.NotBlank;
import io.koraframework.validation.common.annotation.Size;
import io.koraframework.validation.common.annotation.Valid;
import io.koraframework.validation.common.annotation.Validate;

@Root
@Component
public class ResultValidator {

    @Valid
    public record User(@NotBlank String id,
                       @Size(min = 3, max = 6) String name,
                       @Nullable @Size(min = 1, max = 10) String status) {}

    @Valid
    @Validate
    public User create(String name, String status) {
        return new User(UUID.randomUUID().toString(), name, status);
    }
}

File: examples/kotlin/kora-kotlin-validation/src/main/kotlin/io/koraframework/kotlin/example/validation/ResultValidator.kt
package io.koraframework.kotlin.example.validation

import io.koraframework.common.annotation.Component
import io.koraframework.common.annotation.Root
import io.koraframework.validation.common.annotation.*
import java.util.UUID

@Root
@Component
open class ResultValidator {
    @Valid
    @Validate
    open fun create(name: String, status: String?): ResultUser = ResultUser(UUID.randomUUID().toString(), name, status)
}

@Valid
data class ResultUser(
    @field:NotBlank val id: String,
    @field:Size(min = 3, max = 6) val name: String,
    @field:Size(min = 1, max = 10) val status: String?
)

